submit.py

```diff
@@ -470,31 +476,33 @@ def main():
     if not args.force:
         confirm_or_die(problem, language, files, mainclass, tag)
 
-    try:
-        result = submit(submit_url,
-                        login_reply.cookies,
-                        problem,
-                        language,
-                        files,
-                        mainclass,
-                        tag,
-                        args.assignment,
-                        args.contest)
-    except requests.exceptions.RequestException as err:
-        print('Submit connection failed:', err)
-        sys.exit(1)
+    while True:
+        try:
+            result = submit(submit_url,
+                            login_reply.cookies,
+                            problem,
+                            language,
+                            files,
+                            mainclass,
+                            tag)
+        except requests.exceptions.RequestException as err:
+            print('Submit connection failed:', err)
+            sys.exit(1)
 
-    if result.status_code != 200:
-        print('Submission failed.')
-        if result.status_code == 403:
-            print('Access denied (403)')
-        elif result.status_code == 404:
-            print('Incorrect submit URL (404)')
-        else:
-            print('Status code:', result.status_code)
-        sys.exit(1)
+        if result.status_code != 200:
+            print('Submission failed.')
+            if result.status_code == 403:
+                print('Access denied (403)')
+            elif result.status_code == 404:
+                print('Incorrect submit URL (404)')
+            else:
+                print('Status code:', result.status_code)
+            sys.exit(1)
+
+        plain_result = result.content.decode('utf-8').replace('<br />', '\n')
+        if plain_result.startswith('You are out of submission tokens.'): time.sleep(3); continue
```

Comment:
We don't want to auto-resubmit when a user's submission tokens run out (as that could very easily lead to a lot of needless requests).